### PR TITLE
Fix jemalloc compatibility with LLVM libunwind

### PR DESCRIPTION
**Pull request by georgthegreat**
jemalloc provides support for two different libunwind flavors: the original HP libunwind and the one coming with gcc / g++ / libstdc++.

The latter is identified by `JEMALLOC_PROF_LIBGCC` and uses `_Unwind_Backtrace` method instead of `unw_backtrace`.
At the time ClickHouse uses LLVM libunwind which follows libgcc's way of backtracing.

ClickHouse has to provide `unw_backtrace` method by the means of [commit 8e2b31e](https://github.com/ClickHouse/libunwind/commit/8e2b31e766dd502f6df74909e04a7dbdf5182eb1).

While this PR does not allow complete removal of the patch (as ClickHouse itself uses unw_backtrace directly), it definitely sorts some things out.

### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

**Review by alexey-milovidov — APPROVED** [review]

File: contrib/jemalloc-cmake/CMakeLists.txt [hunk 1]
```diff
@@ -170,7 +170,7 @@ endif ()
 target_compile_definitions(_jemalloc PRIVATE -DJEMALLOC_PROF=1)
 
 if (USE_UNWIND)
-    target_compile_definitions (_jemalloc PRIVATE -DJEMALLOC_PROF_LIBUNWIND=1)
+    target_compile_definitions (_jemalloc PRIVATE -DJEMALLOC_PROF_LIBGCC=1)
```
Comment (azat) [inline contrib/jemalloc-cmake/CMakeLists.txt:173]: Actually `unw_backtrace` is used not only in jemalloc, but also in StackTrace.cpp (added in f8d2b5effd0c177f605ce9d27b640753f601ecb6):

https://github.com/ClickHouse/ClickHouse/blob/6d3a54a0440050e531f99976282e6195972a63bb/src/Common/StackTrace.cpp#L305

So this change looks redundant
But it worth to update a comment for `unw_backtrace`

https://github.com/ClickHouse/libunwind/blob/c4ea9848a697747dfa35325af9b3452f30841685/src/libunwind.cpp#L287

Comment (georgthegreat) [reply contrib/jemalloc-cmake/CMakeLists.txt:173]: Indeed, StackTrace.cpp uses unw_backtrace directly.
I think that this usage should be replaced with _Unwind_Backtrace too.
I might proprose a patch later on.

Multiple crutches regarding unw_backtrace() implementation could be removed after this replacement.

**Comment by georgthegreat** [comment]
@alexey-milovidov, @nikitamikhaylov, could you merge this?

**Comment by alexey-milovidov** [comment]
Functional test has failed: https://s3.amazonaws.com/clickhouse-test-reports/37078/6d28b226878f935505edef2606c860da3b58bc9b/stateless_tests__thread__actions__[3/3].html

Obviously, it is not related to the changes in this PR but I will use this fact to highlight the principle that pull requests should be 100% green, regardless. We need to investigate and fix the test and only then merge this PR.

**Comment by alexey-milovidov** [comment]
@mergifyio update

**Comment by alexey-milovidov** [comment]
I merged prematurely, but the code is non-obvious, it needs a comment.

It's literally impossible to understand, why
```
if (USE_UNWIND)
    target_compile_definitions (_jemalloc PRIVATE -DJEMALLOC_PROF_LIBGCC=1)
```
when libunwind is enabled we use "libgcc" for profiling in jemalloc.
It will introduce "WTF" when anyone will read the code.

Need to copy-paste your description from the pull request to the code.

**Pull request merged**